packages/synapse-react-client/src/components/ChallengeTeamWizard/useCreateAndRegisterChallengeTeam.test.ts

```diff
@@ -0,0 +1,293 @@
+const mockInviteUser = jest.spyOn(SynapseClient, 'createMembershipInvitation')
+const mockRegisterForChallenge = jest.spyOn(
+  SynapseClient,
+  'registerChallengeTeam',
+)
+
+const apiError = new SynapseClientError(
+  400,
+  'Something failed',
+  expect.getState().currentTestName!,
+)
+
+function createTeamSuccessMock(teamRequest: CreateTeamRequest): Promise<Team> {
+  return Promise.resolve({
+    ...teamRequest,
+    id: uniqueId(),
+    etag: 'etag',
+    createdOn: new Date().toISOString(),
+    createdBy: String(MOCK_USER_ID),
+    modifiedOn: new Date().toISOString(),
+    modifiedBy: String(MOCK_USER_ID),
+  })
+}
+
+function inviteUserSuccessMock(
+  invitationRequest: CreateMembershipInvitationRequest,
+): Promise<MembershipInvitation> {
+  return Promise.resolve({
+    ...invitationRequest,
+    id: uniqueId(),
+    createdOn: new Date().toISOString(),
+    createdBy: String(MOCK_USER_ID),
+  })
+}
+
+function registerForChallengeSuccessMock(
+  request: CreateChallengeTeamRequest,
+): Promise<ChallengeTeam> {
+  return Promise.resolve({
+    ...request,
+    id: uniqueId(),
+    etag: 'etag',
+  })
+}
+
+const teamName = 'foobar'
+const challengeId = '456789'
+const inviteeEmails = ['foo@bar.com', 'fake@email.net']
+const message = 'Join my team!'
+
+describe('useCreateAndRegisterChallengeTeam', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('creates a team, registers it for a challenge, and sends invitations to join the team', async () => {
+    mockCreateTeam.mockImplementation(createTeamSuccessMock)
+    mockInviteUser.mockImplementation(inviteUserSuccessMock)
+    mockRegisterForChallenge.mockImplementation(registerForChallengeSuccessMock)
+
+    const { result } = render()
+
+    let responses: Awaited<
+      ReturnType<typeof result.current.createAndRegisterTeam>
+    > | null = null
+    await act(async () => {
+      responses = await result.current.createAndRegisterTeam(
+        { name: teamName },
+        challengeId,
+        inviteeEmails,
+        message,
+      )
+    })
+
+    expect(responses).toBeDefined()
+
+    // Check team creation
+    expect(mockCreateTeam).toHaveBeenCalledWith(
+      { name: teamName },
+      MOCK_ACCESS_TOKEN,
+    )
+    const createdTeam = responses![0]
+    expect(createdTeam.name).toBe(teamName)
+
+    // Check challenge registration
+    expect(mockRegisterForChallenge).toHaveBeenCalledWith(
+      {
+        teamId: createdTeam.id,
+        challengeId,
+      },
+      MOCK_ACCESS_TOKEN,
+    )
+    expect(responses![1]).toEqual(
+      expect.objectContaining({
+        teamId: createdTeam.id,
+        challengeId,
+      }),
+    )
+
+    // Check invitations
+    expect(mockInviteUser).toHaveBeenCalledTimes(inviteeEmails.length)
+    for (const inviteeEmail of inviteeEmails) {
+      expect(mockInviteUser).toHaveBeenCalledWith(
+        {
+          teamId: createdTeam.id,
+          inviteeEmail,
+          message,
```

Comment:
Verify that message was included in this request to validate that the bug was fixed